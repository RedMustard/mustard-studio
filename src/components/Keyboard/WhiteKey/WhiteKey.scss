@import '../../../style/variables.scss';

.key--white {
    background-color: #fff;
    border: 1px solid #111;
    box-shadow: 0 1px 3px #999;
    display: table-cell;
    height: $keyboardHeight;
    outline: none;
    width: $whiteKeyWidth;

    &:hover {
        background-color: #eee;
    }

    &:active {
        background-color: #ddd;
    }
}
