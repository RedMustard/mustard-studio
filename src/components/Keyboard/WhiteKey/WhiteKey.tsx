import { h } from 'preact';
import KeyboardKey from '../KeyboardKey/KeyboardKey';

interface WhiteKeyProps {
    onClick: h.JSX.MouseEventHandler<HTMLDivElement>;
}

const WhiteKey = ({ onClick }: WhiteKeyProps) => (
    <KeyboardKey
        className="key--white"
        onClick={onClick}
    />
    // <button
    //     class="key key--white"
    //     onClick={onClick}
    //     type="button"
    //     label="white-key"
    // />
);


export default WhiteKey;
