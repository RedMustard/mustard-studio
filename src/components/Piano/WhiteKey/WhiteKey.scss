@import '../../../style/variables.scss';

.piano-key--white {
    background-color: #f9f9f9;
    border: 1px solid #111;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    display: table-cell;
    height: $pianoHeight;
    outline: none;
    width: $whiteKeyWidth;

    &:hover {
        background-color: #eee;
    }

    &:active {
        background-color: #ddd;
    }
}
