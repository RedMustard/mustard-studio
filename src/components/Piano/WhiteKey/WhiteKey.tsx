import { h } from 'preact';
import { PianoKey } from '../PianoKey/PianoKey';

interface WhiteKeyProps {
    onMouseLeave: (e: MouseEvent) => void;
    onMouseOver: (e: MouseEvent) => void;
    onMouseDown: (e: MouseEvent) => void;
    onMouseUp: (e: MouseEvent) => void;
}

export const WhiteKey = ({
    onMouseLeave,
    onMouseOver,
    onMouseDown,
    onMouseUp,
}: WhiteKeyProps) => (
    <PianoKey
        className="piano-key--white"
        onMouseLeave={(e: MouseEvent) => onMouseLeave(e)}
        onMouseOver={(e: MouseEvent) => onMouseOver(e)}
        onMouseDown={(e: MouseEvent) => onMouseDown(e)}
        onMouseUp={(e: MouseEvent) => onMouseUp(e)}
    />
);
