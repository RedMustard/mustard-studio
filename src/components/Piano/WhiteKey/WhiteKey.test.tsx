import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { WhiteKey } from './WhiteKey';


const baseProps = {
    onMouseDown: jest.fn(),
    onMouseUp: jest.fn(),
    onMouseOver: jest.fn(),
    onMouseLeave: jest.fn(),
};

describe('<WhiteKey />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <WhiteKey
                {...baseProps}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('fires onMouseDown', () => {
        const wrapper = mount(
            <WhiteKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mousedown');
        expect(baseProps.onMouseDown).toBeCalled();
    });

    it('fires onMouseUp', () => {
        const wrapper = mount(
            <WhiteKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseup');
        expect(baseProps.onMouseUp).toBeCalled();
    });

    it('fires onMouseOver', () => {
        const wrapper = mount(
            <WhiteKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseover');
        expect(baseProps.onMouseDown).toBeCalled();
    });

    it('fires onMouseLeave', () => {
        const wrapper = mount(
            <WhiteKey
                {...baseProps}
            />,
        );
        wrapper.find('div').simulate('mouseleave');
        expect(baseProps.onMouseDown).toBeCalled();
    });
});
