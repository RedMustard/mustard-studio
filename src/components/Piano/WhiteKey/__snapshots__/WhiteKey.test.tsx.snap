// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhiteKey /> renders with basic props 1`] = `
<PianoKey
  className="piano-key--white"
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
/>
`;
